types/react-instantsearch-core/index.d.ts

```diff
@@ -771,5 +771,11 @@ export function connectHitInsights(
 ) => React.ComponentType<Omit<ConnectHitInsightsProvided, { insights: WrappedInsightsClient }>>;
 export function connectVoiceSearch(Composed: React.ComponentType<any>): React.ComponentClass<any>;
 
+export interface DynamicWidgetsProps  {
```

Comment:
the dynamic widgets component is also exported from here, so that can be added, as well as connectDynamicWidgets